sympy/matrices/matrices.py

```diff
@@ -190,6 +206,8 @@ def _normalize_op_args(self, op, col, k, col1, col2, error_str="col"):
             if col is None or k is None:
                 raise ValueError("For a {0} operation 'n->kn' you must provide the "
                                  "kwargs `{0}` and `k`".format(error_str))
+            if k == 0:
+                raise ValueError("For a {0} operation, k value must not be 0".format(error_str))
```

Comment:
Although you may state that this is against some linear algebra, 
I feel like this restriction is too artificial and controllable by users, and in fact, this should not always be correcting users theoretically since mathematical zero is not generally decidable.